docs/interface.rst

```diff
@@ -385,9 +385,29 @@ The follow event kinds are understood (see ``status.py`` for more details on the
 
 - ``"download-finished"``: same as upload version.
 
+- ``"invite-created"``: A new invite is created. All invite events also have a ``folder`` indicating which folder they pertain to, a ``id``, ``participant-name`` indicating the invitee and ``mode`` (``"read-only"`` or ``"read-write"``).
+
+- ``"invite-welcomed"``: Connection to the mailbox has succeeded. As well as the keys from ``"invite-created"`` the welcome data is contained in ``welcome`` (you should show users the ``"motd"`` if there is one).
+
+- ``"invite-code-created"``: As well as the keys from ``"invite-created"``, a ``code`` key will be included indicating the wormhole code.
+
+- ``"invite-versions"``: As well as the keys from ``"invite-created"``, a ``versions`` key is included indicating the peer's version information.
+
+- ``"invite-succeeded"``: An invite is successful. All the keys from `"invite-created"`.
+
+- ``"invite-failed"``: An invite has failed. All the keys from `"invite-created"` plus a ``reason``, a freeform string indicating why.
+
+- ``"invite-rejected"``: An invite has been rejected by the other side. All the keys from `"invite-created"` plus a ``reason`` if the other side indicated one.
+
+- ``"invite-cancelled"``: An invite has been cancelled. All the keys from `"invite-created"`.
+
 All timestamps are "seconds since the Unix epoch", as numbers (JSON only has "numbers" and doesn't distinguish floats from ints).
 
 Note that the first "update events" message received will _not_ contain all the updates to that point; it will synthesize the correct events to communicate the current state.
 For example, if there are 50 files in the folder and 48 have already been uploaded, there will be just 2 ``upload-queued`` events (because the other 48 have all finished already).
 If one of these files is currently being uploaded, there will also be a ``upload-started`` event.
 To know the state of all files, use the other endpoints.
+
+
+The *invite events* follow an ordered pattern: ``invite-created`` will be followed by ``invite-created``, ``invite-welcomed``, ``invite-code-created`` and then finalized by one of the "terminal" events (``invite-succeeded``, ``invite-failed``, ``invite-rejected`` or ``invite-cancelled``).
```

Comment:
>``invite-created`` will be followed by ``invite-created``

This seems like an erroneous duplicate. Should the version-related event-kind be included here?

Reply:
yes, that should be whatever you want to call `invite-versions`